common/djangoapps/entitlements/models.py

```diff
@@ -1,8 +1,96 @@
 import uuid as uuid_tools
+from datetime import datetime, timedelta
 
+import pytz
 from django.conf import settings
+from django.contrib.sites.models import Site
 from django.db import models
+
 from model_utils.models import TimeStampedModel
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+class CourseEntitlementPolicy(models.Model):
+    """
+    Represents the Entitlement's policy for expiration, refunds, and regaining a used certificate
+    """
+
+    DEFAULT_EXPIRATION_PERIOD_DAYS = 450
+    DEFAULT_REFUND_PERIOD_DAYS = 60
+    DEFAULT_REGAIN_PERIOD_DAYS = 14
+
+    expiration_period_days = models.IntegerField(
+        default=DEFAULT_EXPIRATION_PERIOD_DAYS,
+        help_text="Number of days from when an entitlement is created until when it is expired.",
+        null=False
+    )
+    refund_period_days = models.IntegerField(
+        default=DEFAULT_REFUND_PERIOD_DAYS,
+        help_text="Number of days from when an entitlement is created until when it is no longer refundable",
+        null=False
+    )
+    regain_period_days = models.IntegerField(
+        default=14,
+        help_text=("Number of days from when an entitlement is redeemed for a course run until "
+                   "it is no longer able to be regained by a user."),
+        null=False
+    )
+    site = models.ForeignKey(Site)
+
+    def get_days_until_expiration(self, entitlement):
+        """
+        Returns an integer of number of days until the entitlement expires
+        """
+        expiry_date = entitlement.created + timedelta(self.expiration_period_days)
+        now = datetime.now(tz=pytz.UTC)
+        return (expiry_date - now).days
+
+    def is_entitlement_regainable(self, entitlement):
+        """
+        Determines from the policy if an entitlement can still be regained by the user, if they choose
+        to by leaving and regaining their entitlement within policy.regain_period_days days from start date of
+        the course or their redemption, whichever comes later, and the expiration period hasn't passed yet
+        """
+        if entitlement.enrollment_course_run:
+            now = datetime.now(tz=pytz.UTC)
+            course_overview = CourseOverview.get_from_id(entitlement.enrollment_course_run.course_id)
+            return (
+                (
+                    (now - course_overview.start).days < self.regain_period_days or
```

Comment:
Will this break if `course_overview.start` is null?

Reply:
From above

>While the column is nullable, in practice as far as I can tell it is a no. I checked on prod

>mysql> SELECT * FROM course_overviews_courseoverview WHERE start IS NULL;
>Empty set (0.02 sec)
>And this is out of 5722 rows.